### ISSUE TITLE
General - Headers - Improve usability of Sub Menu by adding icon to "groups"

### ISSUE THREAD

**Issue by Draise14** [Bforartists/Bforartists#2622]
# Request
![b4a_icons_missing2](https://user-images.githubusercontent.com/35357409/129749341-4292e78a-a279-4d3e-b3cf-30e065c0b075.png)

**Situation:** Some sub menus icons are missing and look incomplete - without icons to find a group of operators means you have to read the "folder" name. There is no iconography. Icons in BFA are key and top-notch, but there still is a little inconsistency with how icons are treated when it comes to folders of operators, or groups of operators. 

This example on Object Mode, "Add" menu, these icons on "folders" or groups or sub-menus already exists and it is a great example how great beneficial it is by having icons also added on all sub-menus that is missing at moment.

With icons+text, improves usability a lot, and it is easy to read, find stuff quickly, memorize, and also is more inviting to the user to explore, finding/discover about Blender/BFA commands, without the need to read texts only (without icons).

**Request**
Add iconography to folders of sub-menus. 

## Reasons
- User knows the meaning of the icon then tend to recognize the menu group faster BEFORE reading
- Faster visual hotspot location of menus for sub-groups of operators (easier to mouse over to). 
- The folder is understood as a folder by the arrow indicator, or could be indicated in the icon also. 
- Has a quick visual overview of the group of operators (without needing to go in to see)
- Would be consistent with the add menu in the 3D View
- Would be consistent with the iconography design philosophy of BFA to have visual indicators over text walls. 

From Pexel, who twitch streams and uses BFA professionally and has done a good campaign to pitch the solution and to why - thanks!:

> Thanks for the support! 
> 
> Although I do think having a indicator for the icons that are folder/sub-menus is great, I don't think necessary needs "over complicate" things regarding new icons in this case of sub-menus.
> If you notice on the image menus, it kinda already has a "indicator" that it is a folder/sub-menu, it is a arrow ">" signal/icon on right side of the command list, that I think it indicates clearly there is a "sub menu" item and having them on right side, it also helps visualise the difference between a "regular command" and when is a "folder/sub-menu command".
>  
> The major "problem" I see here at moment, it is regarding its usability/UX thing, having no icon for all sub-menus commands can makes things a bit bad usability over time, as the users normally needs to read almost all the time the word to find stuff while searching the menu without icons. 
> While if having icons, it helps memorise where it is and then speed up a lot to find stuff via menu.  Plus with icons it encourage/invite the user to discover the commands at glance.
> Also, having missing icons in all sub-menus it makes the menu a bit inconsistent in quality of BFA work, as I think the menu icons it is one of key features difference benefits from using default Blender menus. Anyway it is just a feedback. Hope it makes sense. Thanks! :slight_smile:

**Comment by Draise14** [comment]
This is a very general feature request, so IF this gets the greenlight, I will close this and make a task per editor to update the headers and keep commits granular.

**Comment by ReinerBforartists** [comment]
I think we have talked about this before. That sub menus have no icon is by design. This way these sub menus sticks visually out. There is just one exception, the add menu.

I still want the text menus to stick out. And don't want to over iconize the rest of the menus.

**Comment by Draise14** [comment]
I know we have.. but I'd love to revisit this with you. I am not alone in the need to use iconography even on menu groups. The arrow is the indicator that it's a folder - it would be clear that it sticks out, no?

I did offer the idea that the icon could contain an arrow or folder icon overlay to show that it's a group.
Another alternative is move the arrow, instead of to the right, to the left by the proposed icon. 

_We all just don't want to read so much..._

**Comment by ReinerBforartists** [comment]
Yeah, but my point of view hasn't changed, sorry :)

**Comment by Draise14** [comment]
Is there a compromise somewhere? A middle ground? To be for the artists asking for this?

**Comment by ReinerBforartists** [comment]
What compromise do you think of? I say no and you add it? ^^

I simply see the danger to over iconize the menus here. Icons at operators is great. Icons at menus lets them loose the difference to the operators. And then you start to read again ...

**Comment by Draise14** [comment]
### Minimal idea
Ok, here is a minimal idea from the proposal:
![PureRef_CW9zRo7NzI](https://user-images.githubusercontent.com/35357409/129768121-7fe2e76b-976d-4b26-8aae-bb9638eae969.png)

We use the arrow as indicators to folders. Adding the icon adds visual cues to categorize folders visually AND the way to see folders is with the arrow. 

### Alternative
Arrows before. 
![PureRef_kIfNtslq2O](https://user-images.githubusercontent.com/35357409/129768283-9d1ed7a8-e811-48d5-998c-09c5ccb687a9.png)

Or after.
![image](https://user-images.githubusercontent.com/35357409/129768763-588e1188-9ed2-45e7-a464-a5c0bb750d7f.png)

Here is an idea where we use BOTH icons and the visual cue (the arrows) in the icon and/or move the arrows a little into the title or name to help categorize that that group of operators is an overview to the sub-group. 

If before: the icons offset to show a column category (indent with arrow is folder)
If after: the text indent and arrow (read arrow first) shows folder. 

## TLDR - the problem we're trying to fix in the GUI and UX:
Is this...... there is no visual cue on folders, and over use this adds effort (and time and energy) in a daily UX. 
![image](https://user-images.githubusercontent.com/35357409/129769351-12250d87-84a3-4530-b76d-1561585df350.png)


So.. a compromise is.. find a way to make folder "stick out" and still have some kind of visual cue to visually categorize the folder GUI in headers - and find a balanced consistency.

**Comment by Draise14** [comment]
### Another alternative:
Have an icon that's more a folder and some kind of category icon in it (like in the file browser) to show that it's a sub-folder for operators - but DO NOT use an operator icon, but a new one. 

This means we can visually make folders stick out visually, but not confuse with operator icons.

**Comment by ReinerBforartists** [comment]
Well, rule number one: don't fix what isn't broken. And this is not broken. The icons are not missing. The gaps, or better said the invisible icon indicates that this is a sub menu, and not an operator. This makes it easier to identify the operators in the menu.

To iconize the sub menus like they would be operators would lead to a massive wall of icons. It would make the operators equal sub menus. And INCREASE the search time. This is imho a downgrade in the usability.

The sub menus are usually grouped by separators. So it's not like this would be a big list without visual help where you have to read every entry one by one again.

Mh, we have discussed this one to dead twice already. So sorry to be the bad guy here. But i will not accept this change. Please accept it. Thanks.

![grouping](https://user-images.githubusercontent.com/13919702/129875797-2afac0dd-3084-42d8-8569-04fb6de96c12.jpg)

**Comment by Draise14** [comment]
That is no compromise. No balance. No listening to now a few users and artists who have requested this over time. Not just once, but multiple times. I know you want to dig your toes in and let it be - but let me be the devils advocate here. 

# Trying to understand the reasoning
Let me see if I can understand your reasoning:
1. Too many icons... ? _(... we have 1400 throughout the UI, we pride in iconography, we know how to make this work) If there is a wall of icons, we know how to visually categorize them. Ei, the modifier add list. _
2. You need an invisible icon to show it's a folder? _(We can't even see "invisible", and we have an arrow to show it's a folder? Why do we need something invisible to show a visual cue? It is a hard logic to digest.)_
3. There is less searching because.. we have separators? _From thousands of hours in Bforartists over the years, I still have no choice but to peak into a sub-group to see the operator types? We have to read the titles like in normal Blender? Also, why is it ok to have icons for folders in the Add menu and not in any other? Do we allow for inconsistency? If all folders should be iconless, should we fix the add menu for consistency (hope not!)!?_ 
4. There will be confusion because an operator has an icon, but a folder.. if it were an icon, it would have no distinction from operators? _Even if a proposed (say alternative 3) were a folder symbol to SHOW that it's a folder, not invisibly hint/imply - but explicitly? Even if the UX is currently this: you mouse over a folder, it opens up, clicking on it opens up the folder, and.. it doesn't behave as an operator? There is no way it would be an operator. It has an arrow too! What more distinction is needed? Is this to be standard with other software that don't put icons onto folders? Why do we have icons on folders for blend data types? Can't we be consistent with that?_   

# The Case
If I understand correctly, the reason is this... it's ok to know what groups are what.. by.. glancing at it and seeing an invisible icon.... and nothing else. The only way to find the correct operator group is by.. opening up the sub-menu to peak and/or reading the actual text. At a quick glance, you cannot filter and use visual targeting in the GUI. You have to read the list with the separator position from up to down, count positions, and every line left to right to read the text in zig zag. You either memorize an already read text position and mouse movement to memorize a group of operators by mental count - but even after use, we STILL can't stop zig zag searching to find the operator groups correctly. Muscle memory is not sufficient here. Reading memory is also not efficient. Mental row count also is not useful. 

### Example of the reasoning for no icons
Take this example - according to the original reasoning, it's ok to have an empty space (invisible icon) and the arrow show it's a folder - great... But clearly at a glance, you CAN NOT see what the folder is for. Even with or without separators. Either we memorize row positions, mouse over each group to peak momentarily, or stop and read with zig zag eye movement, slowing down and requiring effort to search. 
![image](https://user-images.githubusercontent.com/35357409/129927809-58f146c9-1231-4853-b274-11442b7e5f1a.png)

This is where we need some kind of compromise and improvement. 

![image](https://user-images.githubusercontent.com/35357409/129931815-0d7c79b7-4e16-466d-b869-3e5c46844b53.png)


## Ideal solution:
Ideally we see the visual cue, target it directly in one Y movement, then go X to the operator we want. Not zig zag eye movement over the text and counting rows and seperators. 

I am all happy with the way separators exist and group things nicely. Great... but according to the reasons.. it's ok to resort to a UX with zigzag reading and peaking, because priority is "no icon" to define that a sub-group is a folder, and this has more priority over what is the folders purpose and GUI peripheral targetting to save time and energy, a counter argument to improve the UX. 

So.. anyway, thanks for the debate - I'll leave these notes here. I hope, with even more user requests for this, we can be sure we're doing the right thing. Are you still sure? 

Can we at least make a test mockup and put it to a commmunity vote?

**Comment by ReinerBforartists** [comment]
No.  I am sure that you will find plenty of users who will vote for this feature. But i have to care for the software as a whole. I will not make a faulty compromise here.

There can just be one UI design. And this means in some cases that i prefer a solution that you don't prefer. Please accept it.

**Comment by Draise14** [comment]
Yeah, I trust you.
